vector/src/main/java/im/vector/VectorApp.java

```diff
@@ -1076,116 +1089,46 @@ public static String localeToLocalisedString(Locale locale) {
+        if (session != null) {
+            mAppAnalytics.visitVariable(7, "Homeserver URL", session.getHomeServerConfig().getHomeserverUri().toString());
+            mAppAnalytics.visitVariable(8, "Identity Server URL", session.getHomeServerConfig().getIdentityServerUri().toString());
         }
-
-        return mPiwikTracker;
-    }
-
-
-    /**
-     * Add the stats variables to the piwik screen.
-     *
-     * @return the piwik screen
-     */
-    private TrackHelper.Screen addCustomVariables(TrackHelper.Screen screen) {
-        screen.variable(1, "App Platform", "Android Platform");
-        screen.variable(2, "App Version", SHORT_VERSION);
-        screen.variable(4, "Chosen Language", getApplicationLocale().toString());
-
-        if (null != Matrix.getInstance(this).getDefaultSession()) {
-            MXSession session = Matrix.getInstance(this).getDefaultSession();
-
-            screen.variable(7, "Homeserver URL", session.getHomeServerConfig().getHomeserverUri().toString());
-            screen.variable(8, "Identity Server URL", session.getHomeServerConfig().getIdentityServerUri().toString());
-        }
-
-        return screen;
     }
 
     /**
      * A new activity has been resumed
      *
      * @param activity the new activity
      */
+
     private void onNewScreen(Activity activity) {
-        if (PreferencesManager.useAnalytics(this)) {
-            Tracker tracker = getPiwikTracker();
-            if (null != tracker) {
-                try {
-                    TrackHelper.Screen screen = TrackHelper.track().screen("/android/" + Matrix.getApplicationName()
-                            + "/" + getString(R.string.flavor_description)
-                            + "/" + SHORT_VERSION
-                            + "/" + activity.getClass().getName().replace(".", "/"));
-                    addCustomVariables(screen).with(tracker);
-                } catch (Throwable t) {
-                    Log.e(LOG_TAG, "## onNewScreen() : failed " + t.getMessage());
-                }
-            }
-        }
+        final String screenPath = "/android/" + Matrix.getApplicationName()
+                + "/" + getString(R.string.flavor_description)
+                + "/" + SHORT_VERSION
+                + "/" + activity.getClass().getName().replace(".", "/");
+        mAppAnalytics.trackScreen(screenPath, null);
     }
 
 
     /**
      * The application is paused.
      */
+
     private void onAppPause() {
-        if (PreferencesManager.useAnalytics(this)) {
-            Tracker tracker = getPiwikTracker();
-            if (null != tracker) {
-                try {
-                    // force to send the pending actions
-                    tracker.dispatch();
-                } catch (Throwable t) {
-                    Log.e(LOG_TAG, "## onAppPause() : failed " + t.getMessage());
-                }
-            }
-        }
+        mAppAnalytics.forceDispatch();
```

Comment:
`PreferencesManager.useAnalytics(this)` not tested anymore (even in `forceDispatch` ?

Reply:
It doesn't make sense to test forceDispatch as we have also recurrent dispatch we can't test. The tests are made when you call tracking methods instead.